addon/services/page-title-list.js

```diff
@@ -190,6 +195,26 @@ export default Service.extend({
     }
   }),
 
+  // TODO use something else for initial render for routes
+  // without model() returning promise
+  // Example: A -> B -> C - when C is the target route and model() takes long time to load
+  // title of A and B will no be rendered until activeTransition.promise resolves.
```

Comment:
This is actually not true and I'll clean this up.  For browser tests there's no difference if we wait for promise.finally or not. For fastboot there is. But since with fastboot for nested routes user anyway sees final result once page is fully returned from server, then it is not possible to see intermediate state. 

And due to waiting for the promise, our update logic for fastboot is now simple title tag adding to head without removing existing tags etc. The previous mess of for loop was required because we eagerly rendered tokens even in mid transitions.